### ISSUE TITLE
New features that would be incredible.

### ISSUE THREAD

**Issue by Leopere** [louislam/uptime-kuma#2245]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Monitor

### 🔖 Feature description

Features from ||REDACTED||
![image](https://user-images.githubusercontent.com/1068374/197548584-d93025c7-1cba-49b1-8b8a-6d2d5adbb45e.png)

 would be incredible... Domain monitoring, DNS monitoring, DNSBL monitoring for mail deliverability.  A lot of this may be vastly out of scope but Uptime-Kuma is pretty incredible currently and if it added these features holy crap it would absolutely be the perfect tool.

### ✔️ Solution

Lose infinite amounts of sleep coding these features into UptimeKuma.

### ❓ Alternatives

There are no alternatives these things are obligatory.

### 📝 Additional Context

Thank you for an awesome tool!

**Comment by rezzorix** [comment]
A lot (if not most / more) of the features the service you are referring to are possible with uptime-kuma.

E.g. checking IPs if they are on a blacklists can be done by using the "HTTP(S) - Keyword"-Monitor. 
No specific "DNSBL" monitor that maybe depends on something else required for that.

Having checked your accounts history it is not an issue, but frankly your post here looks more like an advertising/SEO packed into a "thank you message" rather than a real request :wink:

Anyway; feature-requests are always welcome - however you would need to be more specific.

**Comment by Leopere** [comment]
Oh if its better I can axe the back link it was definitely not attempting to prop up the upstream I was just hoping to get these "premium" features on a system that I use personally.

**Comment by rezzorix** [comment]
Have you run uptime-kuma before?
Except for very few you can do all this with uptime-kuma.

For the 4 additional you listed; theres only 1 that you cannot really do with uptime-kuma yet.

> * [ ]  DNSBL monitor might be useful if its possible however I have no idea how that company who shall not be SEO'd but it would be useful.

Can be done with "HTTP(S) - Keyword" monitor
Several websites / apis available to check that.

> * monitoring domain expiry would be potentially annoying but useful in a scenario when a client or website is not typically under your control directly but you are basically responsible for it.

Can be done with "HTTP(S) - Keyword" monitor
Several websites / apis available to check that.

> * [ ]  DNS monitoring, to determine DNS zone integrity.

Yes that would be nice to have.

> * Port monitoring, would be useful for ensuring non http services are up or making sure sensitive ports are closed to the public.

Can be done with "TCP port" monitor

**Comment by Leopere** [comment]
> Have you run uptime-kuma before? Except for very few you can do all this with uptime-kuma.
> 
> For the 4 additional you listed; theres only 1 that you cannot really do with uptime-kuma yet.
> 
> > * [ ]   DNSBL monitor might be useful if its possible however I have no idea how that company who shall not be SEO'd but it would be useful.
> 
> Can be done with "HTTP(S) - Keyword" monitor Several websites / apis available to check that.
> 
> > * monitoring domain expiry would be potentially annoying but useful in a scenario when a client or website is not typically under your control directly but you are basically responsible for it.
> 
> Can be done with "HTTP(S) - Keyword" monitor Several websites / apis available to check that.
> 
> > * [ ]   DNS monitoring, to determine DNS zone integrity.
> 
> Yes that would be nice to have.
> 
> > * Port monitoring, would be useful for ensuring non http services are up or making sure sensitive ports are closed to the public.
> 
> Can be done with "TCP port" monitor

Realistically I haven't dug into too much of the non default feature set.  I've had Uptime Kuma deployed for a very long time however it upgrades so liquid smoothly that I really haven't had to keep up to date on how much it's changed in the time I've used it for.  So perhaps my knowledge is out of date.  I just have an associate whose leaning into that ||REDACTED|| alternative option and I was thinking gosh Kuma does like 90% of this with the exception of monitoring and alerting on domain expiration and DNS errors.

**Comment by CommanderStorm** [comment]
@Leopere
It seems you forgot to close this issue, as all your problems have been adressed, right?

General feedback:
This issue is not really manageable.
Please open single issues for single features you depend on.
Currently, this issue will end up as an immortal zombie that will never get closed as it is unclear what exactly you want/need.
=> Please close this issue and open clearer issues (1 issue = 1 `feature-request`) of what you need

**Comment by Leopere** [comment]
mostly just wanted feedback and I got it thanks for getting it a touch closer to the grave. 

For the most part it would be really nice to have a few of the features mentioned. if you want to take any of them then great if none of them are within scope then at least I tried.  Makes no difference to me its a great product already granted sometimes it can be a touch on the slower side in certain conditions it doesen't seem happy like if its database is stored on a network volume.  in the end i'm super happy with what y'all have done so far.
